File: Olympiad/COCI/official/2009/contest2_solutions/reseto.cpp
#include <cstdio>

#define MAXN 1000

int n, k, sito[MAXN+1];

int solve() {
   for( int i = 2; i <= n; ++i ) {
      if( sito[i] ) continue;
      for( int j = i; j <= n; j += i ) {
         if( sito[j] ) continue;
         
         if( !--k ) return j;
         sito[j] = 1;
      }
   }
}

int main(void) {
   scanf( "%d%d", &n, &k );
   printf( "%d\n", solve() );
   return 0;
}

File: Olympiad/COCI/official/2009/contest2_solutions/reseto.pas
program reseto;

const MAXN = 1000;

var 
   n, k : longint;
   sito : array[2..MAXN] of boolean;

function solve : longint;
var
   i, j : longint;
begin
   for i := 2 to n do begin
      if sito[i] then continue;
      j := i;
      repeat
         if not sito[j] then begin         
            k := k - 1;
            if k = 0 then solve := j;            
            sito[j] := true;
         end;
         j := j + i;
      until j > n;
   end;
end;

begin
   readln( n, k );
   writeln( solve() );   
end.
